src/EventSubscriber/Init.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\password_enhancements\EventSubscriber;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Session\SessionManagerInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\Url;
+use Drupal\password_enhancements\Controller\UserLogoutController;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Subscribes to Drupal initialization event.
+ */
+class Init implements EventSubscriberInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Messenger.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * Session manager.
+   *
+   * @var \Drupal\Core\Session\SessionManagerInterface
+   */
+  protected $sessionManager;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\user\Entity\User
```

Comment:
Use the interface in the typehint, it is also better to rely on methods required by the contract than the concreate implementation. (Ex.: magic getters are not defined on the interface ;)) The user entity class could be swapped.

`\Drupal\user\UserInterface
`

Reply:
I know, it was by mistake, thanks!